### PR TITLE
Keep CPU architecture in short name (if non-x86).

### PR DESCRIPTION
**Pull request by garloff**
This should address #583

### CODE REVIEW

File: Tests/iaas/flavor-naming/flavor_names.py [hunk 1]
```diff
@@ -234,6 +234,13 @@ def shorten(self):
...
         if self.hype is None and self.hwvirt is None and self.cpubrand is None:
             return self
+        # For non-x86-64, don't strip out CPU brand for short name, as it contains the architecture
+        if self.cpubrand and self.cpubrand.cpuvendor != 'i' and self.cpubrand.cpuvendor != 'z':
+            cpubrand = self.cpubrand
+            # Delete non-relevant details for the short name, only keep brand
+            cpubrand.cpugen = 0
+            cpubrand.perf = ""
+            return Flavorname(cpuram=self.cpuram, disk=self.disk, cpubrand=cpubrand, gpu=self.gpu, ib=self.ib)
         return Flavorname(cpuram=self.cpuram, disk=self.disk, gpu=self.gpu, ib=self.ib)
```
Comment (mbuechse) [inline Tests/iaas/flavor-naming/flavor_names.py:244]: Don't mutate the existing cpubrand object.

```suggestion
        cpubrand = None
        if self.cpubrand and self.cpubrand.cpuvendor not in ('i', 'z'):
            # Delete non-relevant details for the short name, only keep brand
            cpubrand = CPUBrand()
            cpubrand.cpuvendor = self.cpubrand.cpuvendor
            cpubrand.cpugen = 0
            cpubrand.perf = ""
        return Flavorname(cpuram=self.cpuram, disk=self.disk, cpubrand=cpubrand, gpu=self.gpu, ib=self.ib)
```

**Review by mbuechse — COMMENTED** [review]
I think we shouldn't mutate the existing flavor name. Therefore a new instance of `CPUBrand` should be created and used. See my suggestion.

**Comment by garloff** [comment]
Hi Matthias,
I further improved upon your proposal by adding an explicit constructor to the `class CPUBrand`.
Let me know what you think.

File: Tests/iaas/flavor-naming/flavor_names.py [hunk 2]
```diff
@@ -192,6 +192,11 @@ class CPUBrand:
     })
     perf = TblAttr("Performance", {"": "Std Perf", "h": "High Perf", "hh": "Very High Perf", "hhh": "Very Very High Perf"})
 
+    def __init__(self, cpuvendor="i", cpugen=0, perf=""):
```
Comment (mbuechse) [inline Tests/iaas/flavor-naming/flavor_names.py:195]: It's a bit strange that this is the only class of the bunch that has a constructor, which might raise some eyebrow, and it is only used at one occurrence, but I guess these points aren't sufficient to warrant any real objection.

File: Tests/iaas/flavor-naming/flavor_names.py [hunk 3]
```diff
@@ -234,6 +239,11 @@ def shorten(self):
         """return canonically shortened name as recommended in the standard"""
         if self.hype is None and self.hwvirt is None and self.cpubrand is None:
             return self
+        # For non-x86-64, don't strip out CPU brand for short name, as it contains the architecture
+        if self.cpubrand and self.cpubrand.cpuvendor != 'i' and self.cpubrand.cpuvendor != 'z':
```
Comment (mbuechse) [inline Tests/iaas/flavor-naming/flavor_names.py:243]: I still prefer 

```suggestion
        if self.cpubrand and self.cpubrand.cpuvendor not in ('i', 'z'):
```

I would also have preferred the variant with the single call of the constructor, but that's mainly my taste.

File: Tests/iaas/flavor-naming/flavor_names.py (hunk 2 above)
Comment (garloff) [reply Tests/iaas/flavor-naming/flavor_names.py:195]: The fact we don't even need constructors for most is the exception I think, not that we have one here.
The c'tor does what I would expect from a c'tor, so defining it and using it makes the code where the logic happens more readable IMVHO.

**Comment by garloff** [comment]
> Sorry. I meant start a review and posted single comments instead. My bad. I think this can be merged, once you took note of my comments.

No worries. I merged you `in ('a', 'i')` suggestion, that's more pythonic ...

**Pull request merged**